openfe/tests/setup/test_network.py

```diff
@@ -0,0 +1,239 @@
+from dataclasses import dataclass
+from typing import Iterable
+import pytest
+
+from openfe.setup import Molecule, AtomMapping, Network
+from rdkit import Chem
+
+from networkx import NetworkXError
+
+
+@dataclass
+class _NetworkTestContainer:
+    """Container to facilitate network testing"""
+    network: Network
+    nodes: Iterable[Molecule]
+    edges: Iterable[AtomMapping]
+
+    @property
+    def n_nodes(self) -> int:
+        return len(self.nodes)
+
+    @property
+    def n_edges(self) -> int:
+        return len(self.edges)
```

Comment:
This seems a little overengineered, a nametuple would probably work just as well